<a href="https://www.kaggle.com/code/patkim3480/patrick-kariuki-cybershujaa?scriptVersionId=264158473" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

**import the pandas and numpy**

NAME:Patrick Kariuki
TASK: DATA WRANGLING 
DESCRIPTION: DATA WRANGLING USING PYTHON  ON netflix

In [1]:
#import the pandas and numpy
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)



In [2]:
import os
#Defines a variable called filepath and stores the path to the CSV file that contains Netflix shows data
filepath = "/kaggle/input/netflix-shows/netflix_titles.csv"
#Uses Pandas (pd) to read the CSV file at filepath
netflix_df=pd.read_csv(filepath)

# **Discover the Dataset - understand the data**# 

In [3]:
#display the first 10 rows of the dataset
#print("Display the first 10 rows\n\n", df.head(10) )
netflix_df.head(10)

show_id     type                             title  \
0      s1    Movie              Dick Johnson Is Dead   
1      s2  TV Show                     Blood & Water   
2      s3  TV Show                         Ganglands   
3      s4  TV Show             Jailbirds New Orleans   
4      s5  TV Show                      Kota Factory   
5      s6  TV Show                     Midnight Mass   
6      s7    Movie  My Little Pony: A New Generation   
7      s8    Movie                           Sankofa   
8      s9  TV Show     The Great British Baking Show   
9     s10    Movie                      The Starling   

                        director  \
0                Kirsten Johnson   
1                            NaN   
2                Julien Leclercq   
3                            NaN   
4                            NaN   
5                  Mike Flanagan   
6  Robert Cullen, José Luis Ucha   
7                   Haile Gerima   
8                Andy Devonshire   
9                 Theodore Melfi   

                                                cast  \
0                                                NaN   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...   
3                                                NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...   
5  Kate Siegel, Zach Gilford, Hamish Linklater, H...   
6  Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7  Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
8  Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...   
9  Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   

                                             country          date_added  \
0                                      United States  September 25, 2021   
1                                       South Africa  September 24, 2021   
2                                                NaN  September 24, 2021   
3                                                NaN  September 24, 2021   
4                                              India  September 24, 2021   
5                                                NaN  September 24, 2021   
6                                                NaN  September 24, 2021   
7  United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
8                                     United Kingdom  September 24, 2021   
9                                      United States  September 24, 2021   

   release_year rating   duration  \
0          2020  PG-13     90 min   
1          2021  TV-MA  2 Seasons   
2          2021  TV-MA   1 Season   
3          2021  TV-MA   1 Season   
4          2021  TV-MA  2 Seasons   
5          2021  TV-MA   1 Season   
6          2021     PG     91 min   
7          1993  TV-MA    125 min   
8          2021  TV-14  9 Seasons   
9          2021  PG-13    104 min   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   
5                 TV Dramas, TV Horror, TV Mysteries   
6                           Children & Family Movies   
7   Dramas, Independent Movies, International Movies   
8                       British TV Shows, Reality TV   
9                                   Comedies, Dramas   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...  
5  The arrival of a charismatic young priest brin...  
6  Equestria's divided. But a bright-eyed hero be...  
7  On a photo shoot in Ghana, an American model s...  
8  A talented 

In [4]:
#Discover the Dataset - understand the data ;non_null , dtypes, the total columns and the missig values
netflix_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


In [5]:
#get to know the number of the number rows and columns
netflix_df.shape

(8807, 12)

> @the dataframe has 8807 rows and 12 columns****

In [6]:
#list all the columns that  dataframe has  in form oa list
netflix_df.columns.tolist()

['show_id',
 'type',
 'title',
 'director',
 'cast',
 'country',
 'date_added',
 'release_year',
 'rating',
 'duration',
 'listed_in',
 'description']

In [7]:
#Check the datatypes of all the columns
print("The Data types of the various columns\n",netflix_df.dtypes)

The Data types of the various columns
 show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object


In [8]:
#Group and count the missing values using isnull().sum()
print("The Sum of Missing Values per Columns\n\n",netflix_df.isnull().sum())

The Sum of Missing Values per Columns

 show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64


In [9]:
#Group and count the duplicates 
netflix_df.duplicated().sum()

0

# Structuring Format and Standardize 

In [10]:
#format the columns to be consistent 
def clean_columns(netflix_df):
    """
    remove all the whitespaces
    convert the columns to lowercase
    the white spaces in between the words in column with an underscore"""
    netflix_df.columns =   netflix_df.columns.str.strip().str.lower().str.replace(' ', '-')
    return netflix_df
    
#call the clean_cuolumns() function
netflix_df = clean_columns(netflix_df)

#display the the cleaned columns
print(netflix_df.columns)

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in', 'description'],
      dtype='object')


**Datetime conversion**

In [11]:
#convert the date added column to date
netflix_df['date_added'] = pd.to_datetime(netflix_df['date_added'], format='mixed')

In [12]:
netflix_df['date_added'].dtype

dtype('<M8[ns]')

In [13]:
#print(netflix_df['date_added']) 
print(netflix_df['duration'])

0          90 min
1       2 Seasons
2        1 Season
3        1 Season
4       2 Seasons
          ...    
8802      158 min
8803    2 Seasons
8804       88 min
8805       88 min
8806      111 min
Name: duration, Length: 8807, dtype: object


In [14]:
#seprate the duration inot nummeric valuea and unit
netflix_df[['duration_value','duration_unit']] = netflix_df['duration'].str.extract(r'(\d+)\s*(\w+)')
#netflix_df[['duration_value','duration_unit']]

In [15]:
#now convert  hte duration_value to numeric type
netflix_df['duration_value'] = pd.to_numeric(netflix_df['duration_value'], errors='coerce')



In [16]:
#display the results after the conversion
print(netflix_df[['duration_value','duration_unit']])

      duration_value duration_unit
0               90.0           min
1                2.0       Seasons
2                1.0        Season
3                1.0        Season
4                2.0       Seasons
...              ...           ...
8802           158.0           min
8803             2.0       Seasons
8804            88.0           min
8805            88.0           min
8806           111.0           min

[8807 rows x 2 columns]


# CLEANING

**Remove duplicates**

In [17]:
#check for duplicate rows
print("Duplicates\n",netflix_df.duplicated().sum())

Duplicates
 0


In [18]:
netflix_df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

  date_added  release_year rating   duration  \
0 2021-09-25          2020  PG-13     90 min   
1 2021-09-24          2021  TV-MA  2 Seasons   
2 2021-09-24          2021  TV-MA   1 Season   
3 2021-09-24          2021  TV-MA   1 Season   
4 2021-09-24          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  duration_value  \
0  As her father nears the end of his life, filmm...            90.0   
1  After crossing paths at a party, a Cape Town t...             2.0   
2  To protect his family from a powerful drug lor...             1.0   
3  Feuds, flirtations and toilet talk go down amo...             1.0   
4  In a city of coaching centers known to train I...             2.0   

  duration_unit  
0           min  
1       Seasons  
2        Season  
3        Season  
4       Seasons

**Remove /drop the irrelevan columns that will not be useful in the analysis**

In [19]:
#Drop the Description column 
netflix_df= netflix_df.drop(columns=['description'])

In [20]:
#confirm that the column has been dropped 
netflix_df.columns

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in', 'duration_value',
       'duration_unit'],
      dtype='object')

# DATA CLEANING: HANDLING MISSING VALUES


In [21]:
netflix_df.isnull().sum()

show_id              0
type                 0
title                0
director          2634
cast               825
country            831
date_added          10
release_year         0
rating               4
duration             3
listed_in            0
duration_value       3
duration_unit        3
dtype: int64

# ** Handling Missing values for Director column**

In [22]:
#list of the Director Cast pairs and the number of times they apear
netflix_df['dir_cast'] = netflix_df['director']+'---'+ netflix_df['cast']

In [23]:
netflix_df['dir_cast']

0                                                     NaN
1                                                     NaN
2       Julien Leclercq---Sami Bouajila, Tracy Gotoas,...
3                                                     NaN
4                                                     NaN
                              ...                        
8802    David Fincher---Mark Ruffalo, Jake Gyllenhaal,...
8803                                                  NaN
8804    Ruben Fleischer---Jesse Eisenberg, Woody Harre...
8805    Peter Hewitt---Tim Allen, Courteney Cox, Chevy...
8806    Mozez Singh---Vicky Kaushal, Sarah-Jane Dias, ...
Name: dir_cast, Length: 8807, dtype: object

In [24]:
#count the uniques values  in the dir_cast pair
counts = netflix_df['dir_cast'].value_counts()
counts

dir_cast
Rajiv Chilaka---Vatsal Dubey, Julie Tejwani, Rupa Bhimani, Jigna Bhardwaj, Rajesh Kava, Mousam, Swapnil                                                                                                                    12
Rathindran R Prasad---Aishwarya Rajesh, Vidhu, Surya Ganapathy, Madhuri, Pavel Navageethan, Avantika Vandanapu                                                                                                              4
S.S. Rajamouli---Prabhas, Rana Daggubati, Anushka Shetty, Tamannaah Bhatia, Sathyaraj, Nassar, Ramya Krishnan, Sudeep                                                                                                       4
Louis C.K.---Louis C.K.                                                                                                                                                                                                     3
Stan Lathan---Dave Chappelle                                                                           

In [25]:
#Check the dir_cast pair with more than or equal 3 counts
filtered_counts = counts[counts>=3]


In [26]:
#Get the values 
filtered_values = filtered_counts.index

In [27]:
#convert the values into a list
list_dr_cast =list(filtered_values)

#declare a dictionary
dict_direcast = dict()
#
for i in list_dr_cast:
    director, cast  = i.split('---')
    dict_direcast[director] = cast

In [28]:
#Fill in the missing directors
for i in range(len(dict_direcast)):
    netflix_df.loc[(netflix_df['director'].isna()) & (netflix_df['cast'] ==
    list(dict_direcast.items())[i][1]), 'director'] =list(dict_direcast.items())[i][0]

In [29]:
netflix_df.isnull().sum()

show_id              0
type                 0
title                0
director          2616
cast               825
country            831
date_added          10
release_year         0
rating               4
duration             3
listed_in            0
duration_value       3
duration_unit        3
dir_cast          3107
dtype: int64

In [30]:
#Assign 'NotGiven to all other null valuea not filled 
netflix_df.loc[netflix_df['director'].isna(),'director'] = 'Not_Given'

In [31]:
#then confirm the missing values 
netflix_df.isnull().sum()
# Missing values in the Director column successfully filled

show_id              0
type                 0
title                0
director             0
cast               825
country            831
date_added          10
release_year         0
rating               4
duration             3
listed_in            0
duration_value       3
duration_unit        3
dir_cast          3107
dtype: int64

*# Missing values in the Director column successfully filled*

# Missing Values in the Country Column

**Fill in the missing countries**

#Given that there 831 missing countries
#We could use information from the directors to fill in missing countries


In [32]:
#Use directors to fill in the missing countries 
directors = netflix_df['director']
countries = netflix_df['country']

In [33]:
#Pair each director with their country use zip() to get an iterator tuples
pairs = zip(directors , countries)

In [34]:
#Convert the list of tuples inot a dictionary
dir_cntry = dict(list(pairs))

In [35]:
#Director matched to country values used to fill in null country values
for i in range(len(dir_cntry)):
    netflix_df.loc[(netflix_df['country'].isna())&(netflix_df['director'] == 
                                                list(dir_cntry.items())[i][0]),'country']=list(dir_cntry.items())[i][1]

In [36]:
netflix_df.isnull().sum()

show_id              0
type                 0
title                0
director             0
cast               825
country            734
date_added          10
release_year         0
rating               4
duration             3
listed_in            0
duration_value       3
duration_unit        3
dir_cast          3107
dtype: int64

97 missing countries imputed

**#Assign 'Not Given" to all other nulls in the country column**

In [37]:
#Assign 'Not Given" to all other nulls in the country column**
netflix_df.loc[netflix_df['country'].isna(), 'country'] = 'Not_Given'



In [38]:
#Confirm  all the missing values in the country are no filled
netflix_df.isnull().sum()

show_id              0
type                 0
title                0
director             0
cast               825
country              0
date_added          10
release_year         0
rating               4
duration             3
listed_in            0
duration_value       3
duration_unit        3
dir_cast          3107
dtype: int64

**Assign Not_Given to cast nulll**

In [39]:
#Assign not given to cast missing values
netflix_df.loc[netflix_df['cast'].isna(), 'cast'] = 'Not_Given'


# Drop other missing values in the date_added, rating, and duration columnssince they are only a few of them; almost insignificant


In [40]:
#drop other row records that are null

#drop null values in date_added
netflix_df.drop(netflix_df[netflix_df['date_added'].isna()].index, axis=0, inplace=True)

#drop null values in rating 
netflix_df.drop(netflix_df[netflix_df['rating'].isna()].index, axis=0, inplace=True) 

#drop null values in duration 
netflix_df.drop(netflix_df[netflix_df['duration'].isna()].index, axis=0, inplace=True) 

In [41]:
netflix_df.isnull().sum()

show_id              0
type                 0
title                0
director             0
cast                 0
country              0
date_added           0
release_year         0
rating               0
duration             0
listed_in            0
duration_value       0
duration_unit        0
dir_cast          3094
dtype: int64

# Cleaning : Data Inconsistencies

#Date_added columnshould not contain any date value that comes before the 
#release_year, 

In [42]:
#check if there are any added_dates that come before release_year
import datetime as dt

sum(netflix_df['date_added'].dt.year < netflix_df['release_year'])

14

In [43]:
netflix_df.loc[(netflix_df['date_added'].dt.year<netflix_df['release_year']),['date_added', 'release_year']]

date_added  release_year
1551 2020-12-14          2021
1696 2020-11-15          2021
2920 2020-02-13          2021
3168 2019-12-06          2020
3287 2019-11-13          2020
3369 2019-10-25          2020
3433 2019-10-11          2020
4844 2018-05-30          2019
4845 2018-05-29          2019
5394 2017-07-01          2018
5658 2016-12-23          2018
5677 2016-12-13          2017
7063 2018-10-26          2019
7112 2013-03-31          2016

In [44]:
netflix_df.loc[[1551, 1696, 2920, 3168]]

show_id     type          title   director  \
1551   s1552  TV Show          Hilda  Not_Given   
1696   s1697  TV Show   Polly Pocket  Not_Given   
2920   s2921  TV Show  Love Is Blind  Not_Given   
3168   s3169  TV Show   Fuller House  Not_Given   

                                                   cast  \
1551  Bella Ramsey, Ameerah Falzon-Ojo, Oliver Nelso...   
1696  Emily Tennant, Shannon Chan-Kent, Kazumi Evans...   
2920                        Nick Lachey, Vanessa Lachey   
3168  Candace Cameron Bure, Jodie Sweetin, Andrea Ba...   

                                    country date_added  release_year rating  \
1551  United Kingdom, Canada, United States 2020-12-14          2021  TV-Y7   
1696         Canada, United States, Ireland 2020-11-15          2021   TV-Y   
2920                          United States 2020-02-13          2021  TV-MA   
3168                          United States 2019-12-06          2020  TV-PG   

       duration                      listed_in  duration_value duration_unit  \
1551  2 Seasons                       Kids' TV             2.0       Seasons   
1696  2 Seasons                       Kids' TV             2.0       Seasons   
2920   1 Season  Reality TV, Romantic TV Shows             1.0        Season   
3168  5 Seasons                    TV Comedies             5.0       Seasons   

     dir_cast  
1551      NaN  
1696      NaN  
2920      NaN  
3168      NaN

In [45]:
#Confirrm hat no more release_year insconsistences
sum(netflix_df['date_added'].dt.year < netflix_df['release_year'])

14

# CLEANING:ERRORS

Netflix was launched in 1997
There should be no records added to Netflix before this year

In [46]:
#check if there are records dtaed before Netflix 1997
sum(netflix_df['date_added'].dt.year<1997)

0

In [47]:
#Check the unique values for the type column
netflix_df['type'].unique()

array(['Movie', 'TV Show'], dtype=object)

In [48]:
#check the unique value
netflix_df['rating'].unique()

array(['PG-13', 'TV-MA', 'PG', 'TV-14', 'TV-PG', 'TV-Y', 'TV-Y7', 'R',
       'TV-G', 'G', 'NC-17', 'NR', 'TV-Y7-FV', 'UR'], dtype=object)

# TRANSFORMATION

Create separate columns for the listed_in to analyze separately

In [49]:
#check the maximum number of categories for any record for listed column
max(netflix_df['listed_in'].str.split(',').apply(lambda x:len(x)))

#there are a maximum of 3 categories

3

Create three new columns

In [50]:
#splitting the listed_in column and storingthe values in three seprate columns
#if categories <3 , the remaining columns will be filled with null values
netflix_df['listed_in_1'] = netflix_df['listed_in'].str.split(',',expand=True)[0]
netflix_df['listed_in_2'] = netflix_df['listed_in'].str.split(',',expand=True)[1]
netflix_df['listed_in_3'] = netflix_df['listed_in'].str.split(',',expand=True)[2]


In [51]:
netflix_df

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water        Not_Given   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans        Not_Given   
4         s5  TV Show           Kota Factory        Not_Given   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb        Not_Given   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast          country  \
0                                             Not_Given    United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...     South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...  France, Belgium   
3                                             Not_Given        Not_Given   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...            India   
...                                                 ...              ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...    United States   
8803                                          Not_Given        Not_Given   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...    United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...    United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...            India   

     date_added  release_year rating   duration  \
0    2021-09-25          2020  PG-13     90 min   
1    2021-09-24          2021  TV-MA  2 Seasons   
2    2021-09-24          2021  TV-MA   1 Season   
3    2021-09-24          2021  TV-MA   1 Season   
4    2021-09-24          2021  TV-MA  2 Seasons   
...         ...           ...    ...        ...   
8802 2019-11-20          2007      R    158 min   
8803 2019-07-01          2018  TV-Y7  2 Seasons   
8804 2019-11-01          2009      R     88 min   
8805 2020-01-11          2006     PG     88 min   
8806 2019-03-02          2015  TV-14    111 min   

                                              listed_in  duration_value  \
0                                         Documentaries            90.0   
1       International TV Shows, TV Dramas, TV Mysteries             2.0   
2     Crime TV Shows, International TV Shows, TV Act...             1.0   
3                                Docuseries, Reality TV             1.0   
4     International TV Shows, Romantic TV Shows, TV ...             2.0   
...                                                 ...             ...   
8802                     Cult Movies, Dramas, Thrillers           158.0   
8803             Kids' TV, Korean TV Shows, TV Comedies             2.0   
8804                            Comedies, Horror Movies            88.0   
8805                 Children & Family Movies, Comedies            88.0   
8806     Dramas, International Movies, Music & Musicals           111.0   

     duration_unit                                           dir_cast  \
0              min                                                NaN   
1          Seasons                                                NaN   
2           Season  Julien Leclercq---Sami Bouajila, Tracy Gotoas,...   
3           Season                                                NaN   
4          Seasons                                                NaN   
...            ...                                                ...   
8802           min  David Fincher---Mark Ruffalo, Jake Gyllenhaal,...   
8803       Seasons                                                NaN   
8804           min  Ruben Fleischer---Jesse Eisenberg, Woody Harre...   
8805           min  Peter Hewitt---Tim Allen, Courteney Cox, Chevy.

# VALIDATE

In [52]:
#Remove the the dir_cast column created during wrangling
netflix_df.drop(columns=['dir_cast'], inplace=True)

In [53]:
#Validate the number of columns and the datatypes of each
netflix_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8790 entries, 0 to 8806
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   show_id         8790 non-null   object        
 1   type            8790 non-null   object        
 2   title           8790 non-null   object        
 3   director        8790 non-null   object        
 4   cast            8790 non-null   object        
 5   country         8790 non-null   object        
 6   date_added      8790 non-null   datetime64[ns]
 7   release_year    8790 non-null   int64         
 8   rating          8790 non-null   object        
 9   duration        8790 non-null   object        
 10  listed_in       8790 non-null   object        
 11  duration_value  8790 non-null   float64       
 12  duration_unit   8790 non-null   object        
 13  listed_in_1     8790 non-null   object        
 14  listed_in_2     6778 non-null   object        
 15  listed_in

In [54]:
#Display a sample of the wrangled data
netflix_df.sample(7)

show_id     type                                      title  \
1209   s1210    Movie                               Just In Time   
4258   s4259    Movie                         The Birth Reborn 3   
5846   s5847    Movie                                The Do-Over   
2591   s2592    Movie                             The Half Of It   
3793   s3794    Movie                    Svaha: The Sixth Finger   
181     s182  TV Show  Turning Point: 9/11 and the War on Terror   
4182   s4183    Movie                                       Soni   

             director                                               cast  \
1209   Dolapo Adeleke  Sarah Hassan, Stycie Waweru, Mawuli Gavor, Pie...   
4258  Eduardo Chauvet                                          Not_Given   
5846     Steven Brill  Adam Sandler, David Spade, Paula Patton, Kathr...   
2591         Alice Wu  Leah Lewis, Daniel Diemer, Alexxis Lemire, Wol...   
3793    Jang Jae-hyun  Lee Jung-jae, Park Jung-min, Lee Jae-in, Yoo J...   
181         Not_Given                                          Not_Given   
4182         Ivan Ayr  Geetika Vidya Ohlyan, Saloni Batra, Mohinder G...   

            country date_added  release_year rating  duration  \
1209          Kenya 2021-03-12          2020   TV-G    92 min   
4258      Not_Given 2018-12-28          2018  TV-14    73 min   
5846  United States 2016-05-27          2016  TV-MA   109 min   
2591  United States 2020-05-01          2020  PG-13   105 min   
3793    South Korea 2019-05-30          2019  TV-MA   123 min   
181       Not_Given 2021-09-01          2021  TV-14  1 Season   
4182          India 2019-01-18          2018  TV-MA    97 min   

                                             listed_in  duration_value  \
1209            Comedies, Dramas, International Movies            92.0   
4258               Documentaries, International Movies            73.0   
5846                      Action & Adventure, Comedies           109.0   
2591                    Comedies, Dramas, LGBTQ Movies           105.0   
3793                   International Movies, Thrillers           123.0   
181                                         Docuseries             1.0   
4182  Dramas, Independent Movies, International Movies            97.0   

     duration_unit           listed_in_1            listed_in_2  \
1209           min              Comedies                 Dramas   
4258           min         Documentaries   International Movies   
5846           min    Action & Adventure               Comedies   
2591           min              Comedies                 Dramas   
3793           min  International Movies              Thrillers   
181         Season            Docuseries                   None   
4182           min                Dramas     Independent Movies   

                listed_in_3  
1209   International Movies  
4258                   None  
5846                   None  
2591           LGBTQ Movies  
3793                   None  
181                    None  
4182   International Movies

In [55]:
#Reset the row index
netflix_df_cleaned = netflix_df.reset_index(drop=True)

In [56]:
netflix_df_cleaned

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water        Not_Given   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans        Not_Given   
4         s5  TV Show           Kota Factory        Not_Given   
...      ...      ...                    ...              ...   
8785   s8803    Movie                 Zodiac    David Fincher   
8786   s8804  TV Show            Zombie Dumb        Not_Given   
8787   s8805    Movie             Zombieland  Ruben Fleischer   
8788   s8806    Movie                   Zoom     Peter Hewitt   
8789   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast          country  \
0                                             Not_Given    United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...     South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...  France, Belgium   
3                                             Not_Given        Not_Given   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...            India   
...                                                 ...              ...   
8785  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...    United States   
8786                                          Not_Given        Not_Given   
8787  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...    United States   
8788  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...    United States   
8789  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...            India   

     date_added  release_year rating   duration  \
0    2021-09-25          2020  PG-13     90 min   
1    2021-09-24          2021  TV-MA  2 Seasons   
2    2021-09-24          2021  TV-MA   1 Season   
3    2021-09-24          2021  TV-MA   1 Season   
4    2021-09-24          2021  TV-MA  2 Seasons   
...         ...           ...    ...        ...   
8785 2019-11-20          2007      R    158 min   
8786 2019-07-01          2018  TV-Y7  2 Seasons   
8787 2019-11-01          2009      R     88 min   
8788 2020-01-11          2006     PG     88 min   
8789 2019-03-02          2015  TV-14    111 min   

                                              listed_in  duration_value  \
0                                         Documentaries            90.0   
1       International TV Shows, TV Dramas, TV Mysteries             2.0   
2     Crime TV Shows, International TV Shows, TV Act...             1.0   
3                                Docuseries, Reality TV             1.0   
4     International TV Shows, Romantic TV Shows, TV ...             2.0   
...                                                 ...             ...   
8785                     Cult Movies, Dramas, Thrillers           158.0   
8786             Kids' TV, Korean TV Shows, TV Comedies             2.0   
8787                            Comedies, Horror Movies            88.0   
8788                 Children & Family Movies, Comedies            88.0   
8789     Dramas, International Movies, Music & Musicals           111.0   

     duration_unit               listed_in_1              listed_in_2  \
0              min             Documentaries                     None   
1          Seasons    International TV Shows                TV Dramas   
2           Season            Crime TV Shows   International TV Shows   
3           Season                Docuseries               Reality TV   
4          Seasons    International TV Shows        Romantic TV Shows   
...            ...                       ...                      ...   
8785           min               Cult Movies                   Dramas   
8786       Seasons                  Kids' TV          Korean TV Shows   
8787           min                  Comedies            Horror Movies   
8788           min  Children & Family Movies                 Comedi

# PUBLISHING MY DATASET AFTER WRANGLING

**#SAVE MY DATATSET**

In [57]:
#SAVE as CSV
netflix_df_cleaned.to_csv('/kaggle/working/cleaned_netflix.csv', index=False)